solutions/usaco-515.mdx

```diff
@@ -0,0 +1,124 @@
+---
+id: usaco-515
+source: USACO Gold 2015 January
+title: Moovie Mooving
+author: Melody Yu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_movie_gold.html)
+
+## Video Solution
+
+Note: The video solution might not be the same as other solutions. Code in C++.
+<Youtube id="oKm_GZD9u1E" />
+
+## Implementation
+**Time Complexity:** $\mathcal{O}((2^N+CN)N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int N, L;
+vector<vector<int>> schedule;
+vector<vector<vector<int>>> next_show;
+vector<int> length;
+vector<pair<int, int>> dp;
+
+//BeginCodeSnip(Optional Code Snippet Title}
+//bit operations
+inline int pct(int x) { return __builtin_popcount(x); }
+inline int setbit(int n, int k) { return (n | (1 << k)); }
+inline int getbit(int n, int k) { return (n & (1 << k)) >> k; }
+//EndCOdeSnip
```

Comment:
```suggestion
//EndCodeSnip
```

Reply:
oh uh also i don't really care but maybe remove inline? it might confuse some people